### PR TITLE
Feat/password update fix

### PR DESCRIPTION
**Pull request by adamwiteszczak**
Added confirm password param to be sent to api for change password endpoint

### CODE REVIEW

**Review by Smudge3806 — COMMENTED** [review]
Happy with this from a technical perspective... @CallumEddisford is this good from a behaviour expectation?

**Comment by CallumEddisford** [comment]
I usually steer clear of adding loads of paramaters like so:
```
changePassword(userUuid: Xuid<SupportedXuid.User>, newPassword: string, confirmPassword: string, currentPassword: string)
```
This is the way would prefer for now
```
changePassword(userUuid: Xuid<SupportedXuid.User>, params: Object)
```
Really for consistency this should be
```
changePassword(password: Password)
```
where password data:
```
{
    user_id: "xxxx",
    old_password: "old pass",
    password: "new",
    password_confirmation: "new"
}
```

All acomplish the same so happy for what you like, I'd go with the last option where it has a model in SDK

**Comment by adamwiteszczak** [comment]
> I usually steer clear of adding loads of paramaters like so:
> 
> ```
> changePassword(userUuid: Xuid<SupportedXuid.User>, newPassword: string, confirmPassword: string, currentPassword: string)
> ```
> 
> This is the way would prefer for now
> 
> ```
> changePassword(userUuid: Xuid<SupportedXuid.User>, params: Object)
> ```
> 
> Really for consistency this should be
> 
> ```
> changePassword(password: Password)
> ```
> 
> where password data:
> 
> ```
> {
>     user_id: "xxxx",
>     old_password: "old pass",
>     password: "new",
>     password_confirmation: "new"
> }
> ```
> 
> All acomplish the same so happy for what you like, I'd go with the last option where it has a model in SDK

@CallumEddisford  @Smudge3806 I updated the various endpoints and paths to pass an array of values rather than seperate and created a new PasswordChange model in what I think is the correct way. Hope this is better!

**Review by Smudge3806 — DISMISSED** [review]
Good from a technical perspective 👍

**Comment by CallumEddisford** [comment]
`PasswordChange` Model should just be called `Password`?

That model should also have user_id inline with the table

File: src/Services/Auth.ts [hunk 1]
```diff
@@ -255,10 +255,11 @@ class Auth extends Service {
      * @param userUuid
      * @param newPassword
      */
-    changePassword(userUuid: Xuid<SupportedXuid.User>, newPassword: string, currentPassword: string)
+    changePassword(userUuid: Xuid<SupportedXuid.User>, params: PasswordChange)
```
Comment (CallumEddisford) [inline src/Services/Auth.ts:258]:
```suggestion
    changePassword(params: Password)
```

File: src/Services/Auth.ts [hunk 2]
```diff
@@ -255,10 +255,11 @@ class Auth extends Service {
...
     {
         return this.client
-            .patch<AuthResponse>(`1/auth/accounts/${userUuid}/password`, {password: newPassword, current_password: currentPassword})
+            .patch<AuthResponse>(
+                `1/auth/accounts/${userUuid}/password`, params)
```
Comment (CallumEddisford) [inline src/Services/Auth.ts:262]:
```suggestion
                `1/auth/accounts/${params.user_id}/password`, params)
```

File: src/Models/PasswordChange.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Password Change Change
+ */
+export default interface PasswordChange extends Object {
+    current_password: string;
+    password: string;
+    password_confirmation: string;
+}
```
Comment (CallumEddisford) [inline src/Models/PasswordChange.ts:8]: New empty line

File: src/Models/PasswordChange.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Password Change Change
+ */
+export default interface PasswordChange extends Object {
+    current_password: string;
```
Comment (CallumEddisford) [inline src/Models/PasswordChange.ts:5]: user_id

File: src/Models/PasswordChange.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Password Change Change
+ */
+export default interface PasswordChange extends Object {
```
Comment (CallumEddisford) [inline src/Models/PasswordChange.ts:4]: `
```suggestion
export default interface Password extends Object {
```

**Review by CallumEddisford — CHANGES_REQUESTED** [review]

File: src/Models/PasswordChange.ts (hunk 4 above)
Comment (CallumEddisford) [reply src/Models/PasswordChange.ts:5]:
```
user_id: Xuid<SupportedXuid.User>
```

**Comment by adamwiteszczak** [comment]
@CallumEddisford All changes made!

File: docs/content/usage/auth.mdx [hunk 6]
```diff
@@ -474,14 +474,12 @@ Update the password for a user
...
-| userUuid | String | <CheckIcon color="green.5"/> | "92ac7274-9369-11ea-9521-01002da17977" |
-| newPassword | String | <CheckIcon color="green.5"/> | super-secret-password |
-
+| params | Password | <CheckIcon color="green.5"/> | {user_id: '92ac7274-9369-11ea-9521-01002da17977', password: "new-password", password_confirmation: "new-password", current_password: "old-password" }|
 
 **Usage**
 
 ```javascript
-Xedi.Auth.changePassword('92ac7274-9369-11ea-9521-01002da17977', 'super-secret-password');
+Xedi.Auth.changePassword({ user_id: '92ac7274-9369-11ea-9521-01002da17977',  password: "new-password", password_confirmation: "new-password", current_password: "old-password" });
```
Comment (Smudge3806) [inline docs/content/usage/auth.mdx:482]: Might be worth breaking this over a number of lines for readabilities sake #nonblocking

**Review by Smudge3806 — APPROVED** [review]

**Review by CallumEddisford — APPROVED** [review]

**Pull request merged**